src/linear_algebra/nonsingular_inverse.lean

```diff
@@ -406,9 +408,16 @@ end inv
 lemma cramers_rule (A : matrix n n α) (b : n → α) (h : is_unit A.det) :
   cramer A b = A.det • A⁻¹.mul_vec b :=
 begin
-  rw [cramer_transpose_eq_adjugate_mul_vec, A.nonsing_inv_apply h, ← smul_mul_vec_assoc],
+  rw [cramer_eq_adjugate_mul_vec, A.nonsing_inv_apply h, ← smul_mul_vec_assoc],
   conv_rhs { congr, congr, rw ← h.unit_spec, },
   rw units.smul_inv_smul,
 end
 
+/- A stronger form of Cramer's rule that allows us to solve some instances of `A ⬝ x = b` even if
+the determinant is not a unit. A sufficient (but still not necessary) condition is that `A.det`
+divides `b`. -/
+lemma cramers_rule_strong (A : matrix n n α) (b b' : n → α) (h : b = A.det • b') :
```

Comment:
In hindsight, may we should have given `cramers_rule` a "mathlibby" name.
How about `mul_vec_cramer` for this one, and `mul_vec_cramer_of_det_is_unit` for the other one?
I guess you can even leave `h` out of the assumptions, and just substitute `b` everywhere. Then this could become a `simp`-lemma.

Reply:
Agreed, especially on the point about leaving out `h`: what was I thinking!?

I have also updated the `100.yaml` entry to point to the stronger form.